test/e2e/awscodecommitsource/awscodecommitsource.go

```diff
@@ -0,0 +1,206 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awscodecommitsource
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/codecommit"
+	"github.com/aws/aws-sdk-go/service/codecommit/codecommitiface"
+
+	"github.com/triggermesh/aws-event-sources/pkg/apis"
+	apisources "github.com/triggermesh/aws-event-sources/pkg/apis/sources"
+	"github.com/triggermesh/aws-event-sources/pkg/apis/sources/v1alpha1"
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/awscodecommit"
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/conversion"
+	"github.com/triggermesh/aws-event-sources/test/e2e/framework/sources"
+)
+
+var _ = Describe("AWS CodeCommit source", func() {
+	var (
+		srcClient dynamic.ResourceInterface
+		ccClient  codecommitiface.CodeCommitAPI
+
+		sink     duckv1.Destination
+		repoARN  apis.ARN
+		awsCreds credentials.Value
+	)
+
+	f := framework.NewDefaultFramework("awscodecommitsource")
+
+	BeforeEach(func() {
+		srcClient = f.DynamicClient.
+			Resource(conversion.V1alpha1GVR(apisources.AWSCodeCommitSourceResource)).
+			Namespace(f.Namespace.Name)
+
+		By("reading AWS credentials", func() {
+			sess := session.Must(session.NewSession())
+			ccClient = codecommit.New(sess)
+			awsCreds = readAWSCredentials(sess)
+		})
+
+		By("creating a Git repository", func() {
+			arn, undoCreateRepo := awscodecommit.CreateRepository(ccClient, f)
+			framework.AddCleanupAction(
+				undoCreateRepo,
+			)
+			repoARN = parseARN(arn)
+		})
+
+		By("creating an event sink", func() {
+			sink = sources.CreateEventDisplaySink(f.ClientSet, f.Namespace.Name)
+		})
+	})
+
+	It("should generate events on selected actions", func() {
+		By("creating an AWSCodeCommitSource object", func() {
+			src, err := createSource(srcClient, f.Namespace.Name, "test-", sink,
+				withARN(repoARN),
+				withCredentials(awsCreds),
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			sources.WaitForResourceReady(f.DynamicClient, src)
+		})
+
+		By("creating a Git commit", func() {
+			awscodecommit.CreateCommit(ccClient, repoARN.Resource)
+		})
+
+		// TODO(antoineco): check event-display logs, metrics?
```

Comment:
right, how do you check that a valid event was actually received by the event-display ?

Reply:
I think the easiest here would be to read the pod's logs and parse the output to verify that the event conforms to what we expect.

With that being said, I think it's more important to ensure that _an_ event was received in e2e tests, and leave the check of the actual message structure to unit tests.